### DISCUSSION TITLE
Issues with Tilix and VTE

### DISCUSSION THREAD

**Discussion by ghost** [JanDeDobbeleer/oh-my-posh#4674] [Troubleshoot] [answered]
Hello,

After installing oh-my-posh, I had an issue with getting it to init via ~/.bashrc, despite it working perfectly fine as a command. It took me a couple of hours before I realized that one of my other bash configs might be interfering, so after going through the painstaking process of disabling config after config, I noticed that oh-my-posh somehow conflicts with Tilix's [VTE Configuration](https://gnunn1.github.io/tilix-web/manual/vteconfig/).
```
if [ $TILIX_ID ] || [ $VTE_VERSION ]; then
        source /etc/profile.d/vte.sh
fi
```

I read through the discussions [HERE](https://bugzilla.gnome.org/show_bug.cgi?id=697475), [HERE](https://gitlab.gnome.org/GNOME/vte/-/issues/37), and [HERE](https://github.com/UbuntuBudgie/budgie-desktop-environment/issues/105) to try and resolve the issue on my own, but I don't know anything about any of this and this is all just information I tried to understand from reading vte.sh and `oh-my-posh init bash`.

This block in vte.sh should append __vte_osc7 to the PROMPT_COMMAND if it's an array, and __vte_prompt_command if it's not.
```
    if [[ "$(declare -p PROMPT_COMMAND 2>&1)" =~ "declare -a" ]]; then
	PROMPT_COMMAND+=(__vte_osc7)
    else
	PROMPT_COMMAND="__vte_prompt_command"
    fi
```

oh-my-posh init shows that it should prepend the PROMPT_COMMAND if the conditions in the first line are met, but PROMPT_COMMAND then seems to be a string and not an array so I'm guessing that if oh-my-posh is sourced first, vte will override it, but if vte is sourced first, oh-my-posh should just prepend? 

```
if [ "$TERM" != "linux" ] && [ -x "$(command -v /home/gorang/.local/bin/oh-my-posh)" ] && ! [[ "$PROMPT_COMMAND" =~ "_omp_hook" ]]; then
    PROMPT_COMMAND="_omp_hook; $PROMPT_COMMAND"
fi
```

I think I  understand the problem, but this is where I'm stuck. Is there a way for me to get these two to play nicely together?

Thanks for your time and I appreciate any help or suggestions you might have. :)

**Comment by JanDeDobbeleer** [comment, marked as answer]
I looked at VTE's docs and it states this:

> If you use a custom PROMPT_COMMAND instead of simply overriding PS1 you also need to update your PROMPT_COMMAND to append working directory information. This can be achieved by calling __vte_osc7 which gets defined when you source /etc/profile.d/vte-2.91.sh.

That should imply that sourcing the script AFTER oh-my-posh init should preserve the prompt but you'll need to set `"pwd": "osc7"` in oh-my-posh' configuration file.

**Reply by ghost** [reply]
Well now. I overcomplicated the whole thing but I guess it would have helped that I knew what osc7 was when I read the docs the first time. Oh well, TIL. Thanks a lot for your help, it cleared things right up.
